### PR TITLE
Feature/sign in screen

### LINKED ISSUE #17: implement google authentication
Sign into the app using a Google account.

Acceptance Criteria:
- Implement Google Sign-In for authentication.
- After signing in, the user is redirected to the landing page.
- If authentication fails, an error message is displayed.

### PR DESCRIPTION
**Pull request by Ismaillat**
This pull request introduces the SignIn feature and adds UI tests for ensuring the functionality. It also integrates the Google Mobile Services (GMS) plugin in the app-level Gradle file to support Google Sign-In functionality.

Main changes:
Implemented the SignIn feature with Google authentication using Firebase.
Added UI tests to verify the correct behavior of the SignIn screen and Google Sign-In flow.
Integrated the GMS plugin in the app-level build.gradle file to enable Google authentication services.

Closes Issue:

Closes #17

### CODE REVIEW

File: app/src/main/java/com/github/lookupgroup27/lookup/ui/authentication/SignIn.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+@Composable
+fun SignInScreen(navigationActions: NavigationActions) {
+  val context = LocalContext.current
+
+  val launcher =
+      rememberFirebaseAuthLauncher(
+          onAuthComplete = { result ->
+            Log.d("SignInScreen", "User signed in: ${result.user?.displayName}")
+            Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
+            //navigationActions.navigateTo(TopLevelDestinations.MAP) To be implemented
```
Comment (AdrienBousquieEPFL) [inline app/src/main/java/com/github/lookupgroup27/lookup/ui/authentication/SignIn.kt:54]: Mark it as a TODO

File: app/src/main/res/drawable/app_logo.png
Comment (AdrienBousquieEPFL) [inline app/src/main/res/drawable/app_logo.png:1]: We should change the logo with an opacity of 0 behind to make the logo displayable for dark and light theme. We also need one with a white planet and "UP" to adapt for the dark theme.
@othbcq Can you do this ?

**Review by AdrienBousquieEPFL — CHANGES_REQUESTED** [review]
You need to update the branch by merging it with the current state of the main branch and resolve the potential conflicts. By the way, I annotated some things that we can change and things that we can discuss to prepare for modularity with the dark mode. Otherwise, the tests and the code is fine 👍

File: app/src/main/res/drawable/app_logo.png
Comment (othbcq) [reply app/src/main/res/drawable/app_logo.png:1]: Yes, I'll do it

Comment (othbcq) [reply app/src/main/res/drawable/app_logo.png:1]: Here
Design sans titre-22

**Comment by AdrienBousquieEPFL** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=LookUpGroup27_LookUp&pullRequest=23) **Quality Gate failed**
> Failed conditions  [68.5% Coverage on New Code](https://sonarcloud.io/component_measures?id=LookUpGroup27_LookUp&pullRequest=23&metric=new_coverage&view=list) (required ≥ 80%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=LookUpGroup27_LookUp&pullRequest=23)

We will actually skip this for now and add an issue to actually fix this code coverage problem on the SignIn.kt file later.

**Pull request merged**